nerfstudio/data/datamanagers/base_datamanager.py

```diff
@@ -426,22 +440,44 @@ def __init__(
 
         super().__init__()
 
-    def create_train_dataset(self) -> InputDataset:
+    def get_dataset_type(self) -> Type[TDataset]:
+        """Returns the dataset type passed as the generic argument"""
+        orig_class = get_orig_class(self, default_to__class__=True)
+        if orig_class is VanillaDataManager:
+            return TDataset.__default__  # typing: ignore
+        if hasattr(orig_class, "__origin__") and orig_class.__origin__ is VanillaDataManager:
```

Comment:
can we replace the manual access of `__origin__` and `__args__` with `typing.get_origin()` and `typing.get_args()`? or their backports in `typing_extensions`; I imagine it's the same in terms of functionality but might be more succinct + look less intimidating 🙂

Reply:
I have imported them in a new nerfstudio.utils.typing module. Do you agree? After dropping 3.7 we can import directly from typing.

Reply:
Looks good. If we get an OK from @tancik we could also expediting merging for the 3.8 requirement